controllers/helmchart_controller.go

```diff
@@ -456,6 +461,15 @@ func (r *HelmChartReconciler) buildFromHelmRepository(ctx context.Context, obj *
 		}
 	}
 
+	// Try to retrieve the repository index from the cache
+	if r.Cache != nil {
+		if index, found := r.Cache.Get(r.Storage.LocalPath(*repo.GetArtifact())); err == nil {
```

Comment:
The `err` here seems to be related to the `NewChartRepository()` in the above section, which is known to be nil to reach past that code. `err != nil` will result in an error. Is this error check an unrelated mistake?

Reply:
Based on the examples from https://github.com/patrickmn/go-cache, I think it may be related to some old code and the error may not be untended in the current form of the code.

Reply:
thanks @darkowlzz, going to correct this.